policies/releasestrat.html

```diff
@@ -55,65 +56,20 @@ <h5>
 	  fixes. Before that, bug and security fixes will be applied
 	  as appropriate.</p>
 
-	  <p>The next version of OpenSSL will be 1.1.1 which will be an LTS release.
-	  This is currently in development and has a primary focus of implementing
-	  TLSv1.3. The RFC for TLSv1.3 has not yet been published by the IETF.
-	  OpenSSL 1.1.1 will not have its final release until that has happened.</p>
-
-	  <p>The draft release timetable for 1.1.1 is as follows. This may be
-          amended at any time as the need arises.</p>
-
+	  <p>The following stability rules apply to all changes made to code
+	  targetted for a version after 1.1.1:</p>
 	  <ul>
-	    <li>13th February 2018, alpha release 1 (pre1)</li>
-	    <li>27th February 2018, alpha release 2 (pre2)</li>
-	    <li>20th March 2018, beta release 1 (pre3)
-              <ul>
-	        <li>OpenSSL_1_1_1-stable created (feature freeze)</li>
-	        <li>master becomes basis for 1.1.2 or 1.2.0 (TBD)</li>
-	      </ul>
-	    <li>3rd April 2018, beta release 2 (pre4)</li>
-	    <li>17th April 2018, beta release 3 (pre5)</li>
-	    <li>1st May 2018, beta release 4 (pre6)</li>
-	    <li>29th May 2018, beta release 5 (pre7)</li>
-	    <li>19th June 2018, beta release 6 (pre8)</li>
-	    <li>Release readiness check following pre8 release (new release
-	        cycles added if required)</li>
+	    <li>No existing public interface can be removed or modified until its
+	    replacement has been in place in an LTS stable release and the original
+	    function has been documented as deprecated for at least 5 years.</li>
+	    <li>When structures are made opaque, any newly required accessor
+	    macros are added in a feature release of the extant LTS release
+	    and all supported intermediate successor releases (not in security-fix
+	    only mode). Non-trivial accessors may be static inline functions, that
```

Comment:
The text is about making the new accessors available in the **previous** release, where the structures are not yet opaque, and had no accessors.  That is users who're migrating to the new accessors should also find find them in the extant release, rather than everyone having to roll their own.

Reply:
I removed "security fix only mode" from that text.

Reply:
Internally, we've run into instances where we've back-ported accessors and it caused problems with software packages that implemented their own accessors.

Reply:
This would presumably be less of a problem if the upstream accessors were introduced in all our branches simultaneously, saving the other software from needing to ever implement their own.

Reply:
I think if this is clearly stated and followed as policy that accessors are always backported to all live branches, the 3rd party software will handle it fine. The worst thing is to keep the policy undefined - i.e. sometimes backport it and sometimes not or after a long delay, then the 3rd party implementation might really clash.

Reply:
The refusal to provide an officially supported and maintained compatibility header (e.g. `<openssl/compat.h>`) to ease the transition from direct member access to accessor functions was the most significant obstacle for upgrading from 1.0.2 to 1.1.0 IMHO. It is a pity that every downstream project had to reinvent the wheel. (I say this as somebody who is currently porting our company's code from 1.0.2 to 1.1.0...)